garrysmod/lua/autorun/game_hl2.lua

```diff
@@ -1,128 +1,147 @@
-
-local Category = ""
-
-local function ADD_ITEM( name, class, offset )
-
-	list.Set( "SpawnableEntities", class, { PrintName = name, ClassName = class, Category = Category, NormalOffset = offset or 32, DropToFloor = true, Author = "VALVe" } )
-	duplicator.Allow( class )
-
+local spawnableEntities = list.GetForEdit( "SpawnableEntities" )
+local npcWeapons = list.GetForEdit( "NPCUsableWeapons" )
+local playerWeapons = list.GetForEdit( "Weapon" )
+local Allow = duplicator.Allow
+
+local function AddWeapon( className, title, category, npcAllowed, spawnable, author )
+	local weapon = playerWeapons[ className ]
+	if ( weapon == nil ) then
+		weapon = { ClassName = className }
+		playerWeapons[ className ] = weapon
+	end
+
+	weapon.Spawnable = spawnable ~= false
+	weapon.Author = author or "VALVe"
+	weapon.Category = category
+	weapon.PrintName = title
+	Allow( className )
+
+	if npcAllowed then
+		for index = 1, #npcWeapons do
+			local data = npcWeapons[ index ]
+			if ( data and data.class == className ) then
+				data.category = category
+				data.title = title
+				return
+			end
+		end
+
+		npcWeapons[ #npcWeapons + 1 ] = {
+			class = className,
+			category = category,
+			title = title
+		}
+	end
 end
 
-local function ADD_WEAPON( name, class )
-
-	list.Set( "Weapon", class, { ClassName = class, PrintName = name, Category = Category, Author = "VALVe", Spawnable = true } )
-	duplicator.Allow( class )
-
+local function AddEntity( className, title, category, offset, author )
+	local entity = spawnableEntities[ className ]
+	if ( entity == nil ) then
+		entity = { ClassName = className, DropToFloor = true }
+		spawnableEntities[ className ] = entity
+	end
+
+	entity.NormalOffset = offset or 32
+	entity.Author = author or "VALVe"
+	entity.Category = category
+	entity.PrintName = title
+	Allow( className )
 end
 
-Category = "Half-Life 2"
+local category = "Half-Life 2"

```

Comment:
Is it working with Categories? If so, please add them in file for Chinese localizators.

Reply:
In Chinese is possible to localize game titles?!

Reply:
yeah, i guess so https://zh.wikipedia.org/wiki/半衰期2

Reply:
I rechecked, and I don't think it makes sense, because 100% of the other code uses static "Half-Life 2" string, which can be confusing. This requires a more complex solution than replacing a single category name.. I will continue working on game localization, I think I will let you know when a complete and quality localization of the game will be possible.